lib/full-site-editing/template-loader.php

```diff
@@ -65,6 +65,17 @@ function gutenberg_override_query_template( $template, $type, array $templates =
 	global $_wp_current_template_content;
 	$current_template = gutenberg_resolve_template( $type, $templates );
 
+	// Allow falling back to a PHP template if it has a higher priority than the block template.
+	$current_legacy_template = basename( $template, '.php' );
+	$block_template_slug     = is_object( $current_template ) ? $current_template->slug : false;
+	foreach ( $templates as $item ) {
+		$legacy_slug = gutenberg_strip_php_suffix( $item );
+
+		if ( $block_template_slug !== $current_legacy_template && $legacy_slug === $current_legacy_template ) {
+			return $template;
+		}
+	}
+
```

Comment:
Howdy. This whole hierarchy code is pretty complicated so I was thinking if we can "massage" the variable names a bit to make it clearer what is happening :)

```
	// Allow falling back to a PHP template if it has a higher priority than the block template.
	$current_template_slug = basename( $template, '.php' );
	$current_block_template_slug     = is_object( $current_template ) ? $current_template->slug : false;
	foreach ( $templates as $template_item) {
		$template_item_slug = gutenberg_strip_php_suffix( $template_item);

		// Don't override the template if we find a template matching the slug we look for
		// and which does not match a block template slug
		if ( $current_template_slug  !== $current_block_template_slug && $current_template_slug === $template_item_slug) {
			return $template;
		}
	}
```

This may be personal preference, so let me know if it doesn't add up 👍

Reply:
Makes sense.... Done in https://github.com/WordPress/gutenberg/pull/29026/commits/0878b04efbe1b6e5264d5e2649f9aa85dc6ec20b and also ported to https://github.com/WordPress/gutenberg/pull/29298 :+1: